affirm/src/main/java/com/affirm/android/AffirmPlugins.java

```diff
@@ -110,6 +115,11 @@ synchronized AffirmHttpClient restClient() {
                 builder.addHeader("Affirm-User-Agent", "Affirm-Android-SDK");
                 builder.addHeader("Affirm-User-Agent-Version", BuildConfig.VERSION_NAME);
 
+                if (privateKey() != null) {
+                    final String basic = Credentials.basic(publicKey(), privateKey());
+                    builder.addHeader("Authorization", basic);
+                }
```

Comment:
This seems dangerous. I would imagine a private key is not to be used by a client (only by servers) because then it would stop being private. I'm not sure what an attacker could do with such key.

cc: @Frans @empti @yelenaar

Reply:
+1 to @hectorups comment. @wangdaliu the idea here was to return the user back to the "enter the amount" screen and let them go through checkout to cancel out the previously generated card+checkout. We don't want the private key to be added in the SDK client.

Reply:
Hi @diptib10 Does you means that the API `/api/v2/cards/checkout_id/cancel` is unnecessary, just dismiss the UI directly. As the last checkout will be cancelled if the flow is restarted.

Reply:
Hi @diptib10 Can you help confirm that Is my understanding correct? Thanks!

Reply:
@wangdaliu We are still discussing internally to see if there is a secure way to pass the private key during runtime or store it in a hash so it's not easily available in the app. Have you explored any of this on your end? If there isn't an easy way, then just closing the 'vcn display screen' would be the way to go. cc @vcello

Reply:
There is no way to pass a private key into a client. At most, we can have a callback to the merchant client so that the merchant client communicates with the server (they validate the user since it should be signed in) and from the server they use the private key to cancel the vcn.